### PR TITLE
[IOTDB-4185] Prevent RatisConsensus Start Race

### PR DESCRIPTION
**Pull request by SzyWilliam**
Race condition detected as follows:
1. PartitionRegionStateMachine is initialized.
2. ConsensusManager constructor is called, but not returned.
3. Inside ConsensusManager ctor, RaftServer is initialized in ConsensusManager and start election.
4. Inside ConsensusManager ctor, RaftServer is elected and notifies PartitionRegionStateMachine.
5. PartitionRegionStateMachine starts heartbeat service, which uses ConsensusManager.
6. However, in Step5, ConsensusManager Constructor is not returned, which means the ref is null. It can cause NPE

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java [hunk 1]
```diff
@@ -432,14 +433,18 @@ public void startHeartbeatService() {
...
-    if (getConsensusManager().isLeader()) {
-      // Generate HeartbeatReq
-      THeartbeatReq heartbeatReq = genHeartbeatReq();
-      // Send heartbeat requests to all the registered DataNodes
-      pingRegisteredDataNodes(heartbeatReq, getRegisteredDataNodes());
-      // Send heartbeat requests to all the registered ConfigNodes
-      pingRegisteredConfigNodes(heartbeatReq, getRegisteredConfigNodes());
-    }
+    Optional.ofNullable(getConsensusManager())
+        .ifPresent(
```
Comment (CRZbulabula) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java:437]: We'd better add annotation about why should we must use Optional.ofNullable.ifPresent here.

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/PartitionManager.java [hunk 2]
```diff
@@ -496,19 +497,23 @@ public String getRegionStorageGroup(TConsensusGroupId regionId) {
...
-            deletedRegionSet.stream()
-                .map(TRegionReplicaSet::getRegionId)
-                .collect(Collectors.toList()));
-        deletedRegionSet.forEach(
-            regionReplicaSet -> removeRegionGroupCache(regionReplicaSet.regionId));
-        SyncDataNodeClientPool.getInstance().deleteRegions(deletedRegionSet);
-      }
-    }
+    Optional.ofNullable(getConsensusManager())
+        .ifPresent(
```
Comment (CRZbulabula) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/PartitionManager.java:501]: We'd better add annotation about why should we must use Optional.ofNullable.ifPresent here.

**Review by CRZbulabula — COMMENTED** [review]
PTAL~

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java (hunk 1 above)
Comment (SzyWilliam) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java:438]: Done

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/PartitionManager.java (hunk 2 above)
Comment (SzyWilliam) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/PartitionManager.java:502]: Done

**Review by CRZbulabula — APPROVED** [review]
LGTM~

**Pull request merged**